### DISCUSSION TITLE
Get 404 when connecting to local ollama

### DISCUSSION THREAD

**Discussion by Mohamed-E-Fayed** [olimorris/codecompanion.nvim#79] [Q&A] [answered]
Hi,

I was trying adapting your installation to my current setup, but I get 404 when CodeCompanion send request to local Ollama setup.

## Steps:

1. ```bash brew install ollama``` # it is working properly.
2. Copy your config.lua to ~/.config/nvim/lua/codecompanion_config.lua
3.  add the following lines to my ~/.config/nvim/init.vim:
 ```vim
" For installation 
call plug#begin()
  Plug 'nvim-lua/plenary.nvim'
  Plug 'nvim-treesitter/nvim-treesitter'

  Plug 'olimorris/codecompanion.nvim'
call plug#end()


" for configurations
lua require('codecompanion').setup()
lua require('codecompanion_config')
lua <<EOF
require("codecompanion").setup({
  strategies = {
    chat = {
      adapter = "ollama",
    },
    inline = {
      adapter = "ollama",
    },
    agent = {
      adapter = "ollama",
    },
  },
})
require('codecompanion').setup({
server = {
    url = "127.0.0.1:11434"
    },
    model = "custom_llama_3.1"
})
EOF


4. 
```

4. Edited code companion_config.lua and replace all adapter = "open", by "ollama". I mention it though I think it is overwritten by configurations at ~/.config/nvim/init.vim as shown above.
5. run from bash ```bash nvim +PlugInstall ```, and it shows successful installation.
6. in a new file, do :CodeCompanion hello

Output on nvim is nothing,  but found on ollama serve logs:
[GIN] 2024/08/02 - 11:47:43 | 404 |     830.916µs |       127.0.0.1 | POST     "/api/chat"

Expected behavior:
when I use bash and curl, like below, it works.
```bash
$ curl -X POST http://127.0.0.1:11434/api/chat -H "Content-Type: application/json" -d '{"api_key": "your-api-key", "model": "custom_llama_3.1", "input": "Hello, World!"}'

```
The ollama server logs:
```txt
[GIN] 2024/08/02 - 12:11:57 | 200 |   1.56818975s |       127.0.0.1 | POST     "/api/chat"

```

Environment:

Mac OS 13.6 Apple Silicon
ollama version is 0.3.0
NVIM v0.10.0

Thank you so much for your support.

**Comment by olimorris** [comment]
Not sure what is going on in your config as you're calling the `setup` function multiple times. The README should have you covered in terms of how you setup an adapter. Ollama will work out of the box if you haven't changed any of its defaults. The `http://127.0.0.1:11434/api/chat` url is baked into the adapter by default.

**Reply by olimorris** [reply]
@Mohamed-E-Fayed the below should work:

```vim
call plug#begin()
 Plug 'nvim-lua/plenary.nvim'
 Plug 'nvim-treesitter/nvim-treesitter'

 Plug 'olimorris/codecompanion.nvim'
call plug#end()

lua <<EOF
require("codecompanion").setup({
 strategies = {
   chat = {
     adapter = "ollama",
   },
   inline = {
     adapter = "ollama",
   },
   agent = {
     adapter = "ollama",
   },
 },
opts = {
    --log_level = "TRACE"
})
EOF
```

Uncomment the log_level and run `checkhealth codecompanion` if you still have no luck.

**Comment by Mohamed-E-Fayed** [comment]
Hi,

Thank you so much for your fast reply.

Unfortunately, no luck yet.

The same behavior of 404 with CodeCompanion, and 200 with curl still there.
However, I've just noticed that there are 2 other lines generated sometimes when running the command:
```txt
[GIN] 2024/08/02 - 18:38:38 | 200 |      61.041µs |       127.0.0.1 | HEAD     "/"
[GIN] 2024/08/02 - 18:38:38 | 200 |    3.076375ms |       127.0.0.1 | GET      "/api/tags"
[GIN] 2024/08/02 - 18:38:38 | 404 |     218.792µs |       127.0.0.1 | POST     "/api/chat"

```

Here is the output of :checkhealth CodeCompanion

```txt
==============================================================================
CodeCompanion: require("CodeCompanion.health").check()

codecompanion.nvim report ~
- Log file: /Users/mohamedfayed/.local/state/nvim/codecompanion.log
- OK plenary.nvim installed
- OK nvim-treesitter installed
- WARNING telescope.nvim not found
- WARNING dressing.nvim not found
- OK curl installed
```
Thank you so much

**Reply by Mohamed-E-Fayed** [reply]
Here is also the code companion log file. The last request was the one I send earlier of ":CodeCompanion hello".


[codecompanion.log](https://github.com/user-attachments/files/16472398/codecompanion.log)

Hope that's useful.

**Reply by olimorris** [reply]
Can you open a bug report and follow the minimal.lua file? I can't recreate this

**Comment by Mohamed-E-Fayed** [comment, marked as answer]
I’ve just did a bug report.

Thank you so much for your support.
